cmd/edit/cluster/cmd.go

```diff
@@ -192,6 +203,31 @@ func run(cmd *cobra.Command, argv []string) {
 		reporter.Errorf("Failed to get AWS creator: %v", err)
 		os.Exit(1)
 	}
+	awsCredentials := &clusterprovider.AWSCredentials{}
+	if cmd.Flags().Changed("update-aws-credentials") {
+		if confirm.Confirm("update AWS credentials for cluster %s", clusterKey) {
+			ok, err := awsClient.ValidateCredentials()
+			if err != nil {
+				reporter.Errorf("Error validating AWS credentials: %v", err)
+				os.Exit(1)
+			}
+			if !ok {
+				reporter.Errorf("AWS credentials are invalid")
+				os.Exit(1)
+			}
+			reporter.Infof("AWS credentials are valid!")
+
+			// Retrieve the credentials value
+			credValue, err := credentials.NewEnvCredentials().Get()
```

Comment:
I think you should use [`NewChainCredentials`](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/#NewChainCredentials) with two providers: `NewEnvCredentials` and `NewSharedCredentials`. This does two things:
1. Gets credentials from either the env variables or the `~/.aws/credentials` file
2. Allows user to pass `--profile` to use a different AWS profile

https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/#SharedCredentialsProvider

Reply:
@vkareh  I changed to take it from awsclient session. please check now